server/models/AssessmentModel/index.js

```diff
@@ -18,4 +23,12 @@ const AssessmentModel = {
   }),
 }
 
+const assessmentSearchQuery = (queryParams) => {
+  const search = 'search'
+  const searchTerm = Object.hasOwn(queryParams, search)
+
+  if (!searchTerm) return {}
+
+  return { name: { $regex: queryParams.search, $options: 'i' } }
+}
```

Comment:
this function looks for a very specific shape of query params. I think it would be better if the controller pulled the search query out of the query params and the model used that to search for records. That way the model function is more flexible and we don't need all consumers of this model to structure their query params in a specific way to use it.

Reply:
[6fc1baf](https://github.com/AMP-SCZ/dpdash/pull/732/commits/6fc1baff6fa8e01f37cfb213bc4c348295cbe636)